src/pages/index.js

```diff
@@ -45,61 +68,177 @@ popupPreviewImage.setEventListeners();
 //PopupWithForm Class
 const popupProfileForm = new PopupWithForm(
   editProfileModal,
-  handleProfileFormSubmit
+  handleProfileFormSubmit,
+  loadingButtonText
+);
+const popupCardForm = new PopupWithForm(
+  editCardModal,
+  handleCardFormSubmit,
+  loadingButtonText
+);
+const popupProfileImageForm = new PopupWithForm(
+  profileImageModal,
+  handleProfileImageFormSubmit,
+  loadingButtonText
 );
-const popupCardForm = new PopupWithForm(editCardModal, handleCardFormSubmit);
 popupProfileForm.setEventListeners();
 popupCardForm.setEventListeners();
+popupProfileImageForm.setEventListeners();
 
-//Section Class
-const section = new Section(
-  { items: initialCards, renderer: renderCard },
-  cardListEl
+//PopupWithDelete Class
+const popupWithDeleteCard = new PopupWithDelete(
+  deleteCardModal,
+  loadingButtonText
 );
-section.renderItems();
+popupWithDeleteCard.setEventListeners();
 
 //UserInfo Class
-const userClass = new UserInfo({
+const userInfo = new UserInfo({
   nameContainer: profileName,
   descriptionContainer: profileDescription,
+  profilePic,
+});
+
+//Api Class
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "1755563a-df02-4ecd-aa91-5c5958bedadf",
+    "Content-Type": "application/json",
+  },
 });
 
+//Section Class
+const section = new Section({ renderer: renderCard }, cardListEl);
+
+api
+  .getInitialCards()
+  .then((cardItems) => {
+    section.renderItems(cardItems);
+  })
+  .catch((err) => console.log(err));
+
+api
+  .getProfileInfo()
+  .then((res) => {
+    // profilePic.src = res.avatar;
+    userInfo.setAvatar(res);
+
+    return userInfo.setUserInfo({
+      nameInput: res.name,
+      descriptionInput: res.about,
+    });
+  })
+  .catch((err) => console.log(err));
+
+// initialCards.forEach((card) => {
+//   api.addCard(card);
+// });
+
 //
 //Functions
 function fillProfileInputs() {
-  const inputs = userClass.getUserInfo();
+  const inputs = userInfo.getUserInfo();
 
   profileNameInput.value = inputs.profileName;
   profileDescriptionInput.value = inputs.profileDescription;
 }
 
 function handleProfileFormSubmit(data) {
-  userClass.setUserInfo({
-    nameInput: data.title,
-    descriptionInput: data.description,
-  });
-
-  popupProfileForm.close();
+  api
+    .updateProfileInfo({
+      name: data.title,
+      about: data.description,
+    })
+    .then(() => {
+      popupProfileForm.showLoading();
```

Comment:
Needs correction: you should start the loading before the api request (fix that everywhere)